lab3/client/src/views/generator.js

```diff
@@ -0,0 +1,105 @@
+export default function (store) {
+	return class GeneratorComponent extends window.HTMLElement {
+		
+		static get observedAttributes() { return ['data-id']};
+		
+		constructor () {
+			super();
+			
+			// Hey Eric! I have no clue if ive done this corret at all, but I am still extremly confused about this whole setup
+			// We have two constructors for different purposes. Im not sure if i implemented the view correctly.
```

Comment:
This is wrong

Reply:
We need to separate out the data from the view. They are two different things.

In example, you wouldn't consider CSS and HTML to contain any sort of data inside. Data are usually passed from somewhere else. This is the same way that the GeneratorComponent is different from the Generator model. Generator model is a business model that handles the specific logic we have for generator like `getCost` function.